include/zephyr/bluetooth/gatt.h

```diff
@@ -296,8 +296,14 @@ struct bt_gatt_attr {
 	 *  satisfied before calling read() or write().
 	 *
 	 *  @sa bt_gatt_discover_func_t about this field.
+	 *
+	 *  @note The `_auto_assigned_handle` flag in this bitfield indicates if the attribute
+	 *  handle was automatically assigned by the stack (1) or manually set by the
+	 *  application (0).
+	 *  This flag must not be modified by the application.
 	 */
-	uint16_t perm;
+	uint16_t perm: 15;
+	uint16_t _auto_assigned_handle: 1;
```

Comment:
@jhedberg correct me if I'm wrong, but I think we want to avoid bitfields in the public header? And prefer to use macros to get and set bits instead. 

IIRC the reasoning is that these type of bitfields may not work well with non-arm architectures, but unsure if we actually have that in writing anywhere in Zephyr

Reply:
Okay. I'm pretty sure I've been asked to not use bitfields in the public header files, even for non-packed structs several times, and mostly for performance reasons since some architectures do not support it. 

Anyways, since we do not (AFAIK) have any coding guidelines on this, it's OK as is :)